XRP Ledger Introduction

mDuo13 opened the pull request:
A much-needed high-level introduction to what the XRP Ledger does, how it's different from other things (especially other cryptocurrencies), and why it's so awesome.

I expect this to become a pretty important document, so lots of feedback is appreciated. (I can only imagine there'll be lots of points of contention in what should be covered at this level.) There are a few places where I'd like to have better docs to link to, but I'm working with what we have right now. (We can update this later as we get more quality intro-level docs written.)

mDuo13 committed:
XRP Ledger Intro (1/?)

mDuo13 committed:
XRP Ledger Intro supply, censorship resistance, etc. (2/?)

mDuo13 committed:
XRP Ledger Intro software gov. (3/?)

mDuo13 committed:
XRP Ledger Intro crypto, features, exchange (4/4, 1st draft complete)

mDuo13 committed:
XRP Ledger Intro: fix broken link

intelliot commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+# XRP Ledger Introduction
+
+The **XRP Ledger** is a decentralized cryptocurrency network. Like all such networks, it's powered by a peer-to-peer server and it enables anyone to send and receive value within the rules of the network. The XRP Ledger is the home of **XRP**, a cryptocurrency which was designed by its creators to bridge the many different currencies preferred by people worldwide. Ripple, the company that steward ...
```
When do we use "cryptocurrency" vs. "digital asset"?

> The **XRP Ledger** is a decentralized cryptographic ledger powered by a network of peer-to-peer servers. It is the home of **XRP**, a digital asset designed to bridge the many different currencies preferred by people worldwide. Ripple, the company that stewards development of the XRP Ledger, expects XRP to be a key part of enabling the _Internet of Value_: a world in which money moves the way information does today.

Above, I've made an attempt to tighten up the wording slightly and avoid using the word "cryptocurrency." Just an idea.

mDuo13 replied:
That's an interesting thought.

"Cryptocurrency" is a specific subset of "digital asset" and has become fairly well-known at this point. The term "digital asset" rightfully includes things like rewards points that are centrally administered and have very different properties than cryptocurrencies.

I don't think that "cryptographic ledger" is any more intuitive than "cryptocurrency network" and I suspect that our target audience understands the latter term better, but sometimes it's hard to know when I'm in a bubble that happens to look at the terms differently.

jbheron replied:
Hmm, we should get aligned on this semantic issue. Brad G. has been pretty public with the use of digital assets, instead of cryptocurrency. I like omitting the word cryptocurrency. I suggest tightening up Elliot's recommendation a little further:

> The **XRP Ledger** is a decentralized cryptographic ledger powered by a network of peer-to-peer servers. It is the home of **XRP**, a digital asset designed to bridge the many different currencies in use, worldwide. Ripple stewards the development of the XRP Ledger and advances XRP as a key contribution to the _Internet of Value_: a world in which money moves the way information does today.

jhaaaa replied:
I attended a Ripple Pitch Training delivered by Asheesh that told us to call XRP a "digital asset," not a "cryptocurrency." Is this a rule that we want to apply to the tech docs? What is the difference between a digital asset and a cryptocurrency? Does digital asset more accurately describe XRP? We may need to sync up with PM on messaging.

bachase commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+With (physical) coins and paper money, individuals can do business without going through a central party. Non-cryptocurrency digital assets all have a central administrator to confirm and validate transactions, who also has the power to censor or roll back transactions, or disallow some individuals from using the digital asset. (For example, if a digital money company decides a person has violate ...
+
+**Note:** Non-XRP currencies issued in the XRP Ledger _can_ be frozen. For more information, see the [Freeze documentation](concept-freeze.html).
```
Maybe add `currencies issued by *users*` to help highlight that Ripple has no special role.?

bachase commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+Alongside war and political turmoil, hyperinflation is one of the leading causes of death for currencies. While the decentralized system of validators provides XRP with some resistance to political factors, the rules of the XRP Ledger provide a simpler solution to hyperinflation: the total supply of XRP is finite. Without a mechanism to create more, it becomes much less likely that XRP could suff ...
+
+The supply of XRP available to the general _does_ change due to a few factors:
```
Missing word `to the general *public*`?

bachase commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+- Regularly running automated checks for potential vulnerabilities and memory leaks
+- Regularly commissioning external reviews by professional organizations
+
+The XRP Ledger also has a built-in "Amendments" system for evolving the existing feature set and transitioning from old rules to new ones. Rather than fracturing the ecosystem or causing uncertainty whenever a transition occurs, any changes to transaction processing in the XRP Ledger are switched on by Amendments that require sustained approval from validators over 2 weeks, giving everyone time t ...
```
Consider breaking this out as its own section. Although it does fall under your `Responsible Software Governance` category, Amendments seem a bit different than the software engineering qualities listed above.

bachase commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+## Smart Features for Smart Contracts
+[Smart Features for Smart Contracts]: #smart-features-for-smart-contracts
+
+Besides simple value transfer via XRP payments, the XRP Ledger has several advanced features that provide useful functions for building applications that use Internet of Value to serve previously unknown or impractical needs. Rather than running applications as "smart contracts" in the network itself, the XRP Ledger provides tools for contracts, while letting the applications themselves run anywh ...
```
Missing `the` before Internet of Value.

bachase reviewed (commented):
👍 Just a few nits. Great addition to highlight the qualities of XRPL

jbheron commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+## The Best Digital Asset
+
+XRP is the native cryptocurrency of the XRP Ledger. Anyone with a cryptographic key and an internet connection can receive, hold, and send XRP to anyone else. From the beginning, XRP's creators have developed it to be a desirable bridge currency which can facilitate trades in any other currency. XRP has many properties which make it an appealing asset for many other use cases, too:
```
Whatever we decide, here's another use of the term cryptocurrency.

jbheron replied:
'From the beginning' jumps out as a temporal hallmark. Also, 'which' => 'that', as the clause is central to the meaning of the sentence. I suggest:
> The creators of XRP developed it to be a desirable bridge currency that can facilitate trades in any other currency. Additionally, the properties of XRP make it an appealing asset for many other use cases:

jbheron commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+XRP is the native cryptocurrency of the XRP Ledger. Anyone with a cryptographic key and an internet connection can receive, hold, and send XRP to anyone else. From the beginning, XRP's creators have developed it to be a desirable bridge currency which can facilitate trades in any other currency. XRP has many properties which make it an appealing asset for many other use cases, too:
+
+- **[Censorship-resistant transaction processing][]:** No single party decides which XRP transactions are approved or not, and no one can "roll back" a transaction after it has happened. As long as those who choose to participate in the network keep it healthy, XRP can be sent and received in seconds.
```
Suggest 'has happened' becomes 'completes', remove quoted roll back, and modify passive voice.

> - **[Censorship-resistant transaction processing][]:** No single party decides which XRP transactions are approved or not, and no one can roll back a transaction after it completes. As long as those who choose to participate in the network keep it healthy, they can send and receive XRP in seconds.

jbheron replied:
🤦‍♂️, I missed one of the passive voice uses. Suggest:
> - **[Censorship-resistant transaction processing][]:** No single party decides which XRP transactions the XRP Ledger Network approves, and no one can roll back a transaction after it completes. As long as those who choose to participate in the network keep it healthy, they can send and receive XRP in seconds.

jbheron commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+XRP is the native cryptocurrency of the XRP Ledger. Anyone with a cryptographic key and an internet connection can receive, hold, and send XRP to anyone else. From the beginning, XRP's creators have developed it to be a desirable bridge currency which can facilitate trades in any other currency. XRP has many properties which make it an appealing asset for many other use cases, too:
+
+- **[Censorship-resistant transaction processing][]:** No single party decides which XRP transactions are approved or not, and no one can "roll back" a transaction after it has happened. As long as those who choose to participate in the network keep it healthy, XRP can be sent and received in seconds.
+- **[Fast, Efficient Consensus Algorithm][]:** The XRP Ledger's consensus algorithm settles transactions in 4 to 5 seconds, processing at a throughput of up to 1500 transactions per second. These properties put XRP orders of magnitude ahead of other top digital assets.
```
Careful, orders of magnitude implies 2 orders, minimum. 1500/10*2 = 15, which is not the maximum TPS for other currencies or payment systems.

MarkusTeufelberger replied:
Also the "up to" is measured in lab conditions (GBit LAN, relatively beefy machines, very simple transactions on a very small ledger state).

mDuo13 replied:
What a coincidence. 15 _is_ the maximum TPS for other top digital assets.
![XRP Portal cryptocurrency asset comparison](https://cdn.ripple.com/wp-content/themes/ripple-beta/assets/img/xrp/scalable@2x.png)

jbheron commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+- **[Censorship-resistant transaction processing][]:** No single party decides which XRP transactions are approved or not, and no one can "roll back" a transaction after it has happened. As long as those who choose to participate in the network keep it healthy, XRP can be sent and received in seconds.
+- **[Fast, Efficient Consensus Algorithm][]:** The XRP Ledger's consensus algorithm settles transactions in 4 to 5 seconds, processing at a throughput of up to 1500 transactions per second. These properties put XRP orders of magnitude ahead of other top digital assets.
+- **[Finite XRP Supply][]:** When the XRP Ledger began, 100 billion XRP were created, and no more XRP can be created after that. (Each XRP is subdivisible down to 6 decimal places, for a grand total of 100 quintillion "drops" of XRP.) The available supply of XRP decreases slowly over time as small amounts are destroyed to pay transaction costs.
```
Temporal suggestion, also are new phrases italicized? If so:
> - **[Finite XRP Supply][]:** When the XRP Ledger began, 100 billion XRP were created, and no more XRP will ever be created. (Each XRP is subdivisible down to 6 decimal places, for a grand total of 100 quintillion _drops_ of XRP.) Transaction costs for using the network destroy the available supply of XRP, which decreases slowly over time.

mDuo13 replied:
I'm leaving "small amounts are destroyed to pay transaction costs" in passive voice by design, so that the sentence can start with the more important "The supply of XRP decreases slowly over time" clause.

jhaaaa commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+The **XRP Ledger** is a decentralized cryptocurrency network. Like all such networks, it's powered by a peer-to-peer server and it enables anyone to send and receive value within the rules of the network. The XRP Ledger is the home of **XRP**, a cryptocurrency which was designed by its creators to bridge the many different currencies preferred by people worldwide. Ripple, the company that steward ...
+
+## The Best Digital Asset
```
Same question as above -- if we want to address it -- do we want to refer to XRP as a digital asset or a cryptocurrency? Or is it both - and we refer to it as a digital asset in some contexts and as a cryptocurrency in others? What are those contexts?

jhaaaa commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+XRP is the native cryptocurrency of the XRP Ledger. Anyone with a cryptographic key and an internet connection can receive, hold, and send XRP to anyone else. From the beginning, XRP's creators have developed it to be a desirable bridge currency which can facilitate trades in any other currency. XRP has many properties which make it an appealing asset for many other use cases, too:
+
+- **[Censorship-resistant transaction processing][]:** No single party decides which XRP transactions are approved or not, and no one can "roll back" a transaction after it has happened. As long as those who choose to participate in the network keep it healthy, XRP can be sent and received in seconds.
```
micro-nit 😂 : initial cap the text in bold - to match the init-cap case pattern used for the others?

jhaaaa commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+## Fast, Efficient Consensus Algorithm
+[Fast, Efficient Consensus Algorithm]: #fast-efficient-consensus-algorithm
+
+The XRP Ledger's biggest difference from most cryptocurrencies is that it uses a unique consensus algorithm that does not waste time and energy on "mining", the way Bitcoin, Etherium, and almost all other such systems do. Instead of "proof of work" or even "proof of stake", The XRP Ledger's consensus algorithm uses a system where every participant has an overlapping set of "trusted validators" an ...
```
Change "Etherium" to "Ethereum"

Change "of stake", The XRP Ledger's" to "of stake", the XRP Ledger's"

Consider changing "wastes more electricity" to "uses more electricity"

MarkusTeufelberger replied:
"A single Bitcoin transaction wastes more electricity than a family home in the USA uses in an entire day" - this is misleading and/or wrong, the same goes for the second part of that sentence. It is also irrelevant to an explanation about how XRPL works.

mDuo13 replied:
Respectfully, the fact that the XRP Ledger does not incentivize a competition for who can muster the most computing power is a _very_ relevant part of how the XRP Ledger works. It motivates core design decisions, is the _biggest_ differentiator from proof-of-work cryptocurrencies, and is potentially very important for the future of the planet.

I've rephrased the first clause to be less incendiary, though:

> As of early 2018, the amount of electricity the Bitcoin network uses per transaction is more than a family home in the USA uses in an entire day, and confirming the transaction takes hours.

MarkusTeufelberger replied:
"estimated to use" - and the usage is per block, not per transaction. Confirmation also doesn't take hours, it doesn't even take a single hour on average (even for 6 blocks, which is already something I would only require for _very_ high value transactions)

jhaaaa commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+A sample of advanced features in the XRP Ledger:
+
+- [Payment Channels](tutorial-paychan.html) allow asynchronous balance changes as fast as you can create and validate signatures.
+- [Escrow](concept-escrow.html) locks up XRP until a dedicated time or cryptographic condition.
```
Consider changing "until a dedicated time or cryptographic condition." to "until a dedicated time or a cryptographic condition is met."

MarkusTeufelberger replied:
until a dedicated [point in] time [has passed] or [a] cryptographic condition [is revealed].

MarkusTeufelberger commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+
+The XRP Ledger's biggest difference from most cryptocurrencies is that it uses a unique consensus algorithm that does not waste time and energy on "mining", the way Bitcoin, Etherium, and almost all other such systems do. Instead of "proof of work" or even "proof of stake", The XRP Ledger's consensus algorithm uses a system where every participant has an overlapping set of "trusted validators" an ...
+
+For more information on how the XRP Ledger's consensus algorithm works, see [The XRP Ledger Consensus Process](concept-consensus.html). For background on why the XRP Ledger uses a consensus algorithm, see [Reaching Consensus In the XRP Ledger](concept-reaching-consensus.html).
```
"why the XRP Ledger uses a consensus algorithm" - maybe you meant "why the XRP Ledger uses **this** consensus algorithm"? It is quite clear why some consensus algorithm is being used...

jhaaaa commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+- [Payment Channels](tutorial-paychan.html) allow asynchronous balance changes as fast as you can create and validate signatures.
+- [Escrow](concept-escrow.html) locks up XRP until a dedicated time or cryptographic condition.
+- [DepositAuth](concept-depositauth.html) lets an address decide who can send it money and who can't.
+- And of course, there's the [decentralized exchange](#on-ledger-decentralized-exchange) for trading obligations on-ledger...
```
Is this ellipsis intentional? Just checking. Adding some suspense and drama? 😺

MarkusTeufelberger replied:
Maybe adding a gif of https://www.youtube.com/watch?v=a1Y73sPHKxw might be in order?

mDuo13 replied:
Yes, the ellipsis is intentional. I'm alluding to the fact that the decentralized exchange is kind of a massive suite of functionality built into the XRP Ledger which isn't even core to all the other big differences.

I _could_ change this bullet point to be more workmanlike, for example:

> - A [Decentralized Exchange] for trading obligations on-ledger.

... but I don't _wanna_. :tired_face:

MarkusTeufelberger commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+- A famously strict and thorough code review process
+- Comprehensive code coverage and unit tests
+- Regularly running automated checks for potential vulnerabilities and memory leaks
+- Regularly commissioning external reviews by professional organizations
```
Which are published at...?

mDuo13 replied:
The standard for this type of review is that it's confidential. My (non-lawyer) understanding is that we _legally can't_ release the reports we've commissioned in the past.

MarkusTeufelberger replied:
Hm, so Ripple has to be trusted that these secret reviews by unnamed "professional organizations" are actually taking place. Here's a project that did similar stuff in the open (after all, it is for open source software): https://opencryptoaudit.org

Maybe you could mention the bug bounty program here too?

Another thing that's missing (but out of scope for this description likely) is a public process to get features added, I have yet to see any major contribution to rippled from outside the company. I suspect one of the reasons is that new features are discussed on JIRA and only the code ends up in the open. Something like a CONTRIBUTING file can go a long way.

jbheron commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+- **[Censorship-resistant transaction processing][]:** No single party decides which XRP transactions are approved or not, and no one can "roll back" a transaction after it has happened. As long as those who choose to participate in the network keep it healthy, XRP can be sent and received in seconds.
+- **[Fast, Efficient Consensus Algorithm][]:** The XRP Ledger's consensus algorithm settles transactions in 4 to 5 seconds, processing at a throughput of up to 1500 transactions per second. These properties put XRP orders of magnitude ahead of other top digital assets.
+- **[Finite XRP Supply][]:** When the XRP Ledger began, 100 billion XRP were created, and no more XRP can be created after that. (Each XRP is subdivisible down to 6 decimal places, for a grand total of 100 quintillion "drops" of XRP.) The available supply of XRP decreases slowly over time as small amounts are destroyed to pay transaction costs.
+- **[Responsible Software Governance][]:** The XRP Ledger's software is maintained by a team of world-class developers at Ripple, who maintain the security of the network while adding features and stewarding future developments. Ripple acts as a steward for the software and an advocate for its interests, so the XRP Ledger ecosystem has a constructive relationship with governments and financial in ...
```
A team which maintains, not who maintains-- also, passive voice. So, a little re-wording could strengthen and slim it. Here's my try:

> Ripple's team of world-class developers maintain the software that powers the XRP Ledger. The developers secure the network, and promote future developments by adding new features. Ripple acts as a steward for the software and an advocate for its interests. In this way, the XRP Ledger ecosystem builds a constructive relationship with governments and financial institutions.

intelliot replied:
I feel uneasy with the idea that the developers secure the network. Perhaps:

> The developers increase the value of the XRP Ledger by regularly improving its technology and adding new features.

mDuo13 replied:
Fair points. Revised as follows:

> A team of full-time, world-class developers at Ripple maintain and continually improve the XRP Ledger's underlying software. Ripple acts as a steward for the technology and an advocate for its interests, and builds constructive relationships with governments and financial institutions worldwide.

jbheron commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+- **[Finite XRP Supply][]:** When the XRP Ledger began, 100 billion XRP were created, and no more XRP can be created after that. (Each XRP is subdivisible down to 6 decimal places, for a grand total of 100 quintillion "drops" of XRP.) The available supply of XRP decreases slowly over time as small amounts are destroyed to pay transaction costs.
+- **[Responsible Software Governance][]:** The XRP Ledger's software is maintained by a team of world-class developers at Ripple, who maintain the security of the network while adding features and stewarding future developments. Ripple acts as a steward for the software and an advocate for its interests, so the XRP Ledger ecosystem has a constructive relationship with governments and financial in ...
+- **[Secure, Adaptable Cryptography][]:** The XRP Ledger relies on industry standard digital signature systems like ECDSA (the same scheme used by Bitcoin) but also supports modern, efficient algorithms like Ed25519. The extensible nature of the XRP Ledger's software makes it possible to add and disable algorithms as the state of the art in cryptography advances.
+- **[Smart Features for Smart Contracts][]:** The XRP Ledger has features like Escrow, Checks, and Payment Channels to support building cutting-edge financial applications, especially the [Interledger Protocol](https://interledger.org/). This toolbox of advanced features comes with separate double-checks on invariant rules, so even a glitch in the transaction implementation won't cause the XRP Le ...
```
To improve readability:
> Features like Escrow, Checks, and Payment Channels support the creation of cutting-edge financial applications, such as the [Interledger Protocol](https://interledger.org/). This toolbox of advanced features comes with separate double-checks on invariant rules, so even a glitch in the transaction implementation won't cause the XRP Ledger to become inconsistent or corrupt.

jbheron commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -0,0 +1,88 @@
+## Fast, Efficient Consensus Algorithm
+[Fast, Efficient Consensus Algorithm]: #fast-efficient-consensus-algorithm
+
+The XRP Ledger's biggest difference from most cryptocurrencies is that it uses a unique consensus algorithm that does not waste time and energy on "mining", the way Bitcoin, Etherium, and almost all other such systems do. Instead of "proof of work" or even "proof of stake", The XRP Ledger's consensus algorithm uses a system where every participant has an overlapping set of "trusted validators" an ...
```
Cryptocurrencies use.

jhaaaa commented:
Added just a few comments. Concisely summarizing what a product does to enable users to decide whether to adopt or contribute to it---is a monumental task. I think you are one of the few who could have done it using such user-friendly language, while also delivering the key technical details. 👍

jbheron reviewed (commented):
I've done a partial review, here, and want to stop to better understand the audience, a few other details, and to give room for general editing.

Much of this introduction seems to be in somewhat of a marketing style, or technical marketer language. Is this intended? I'm somewhat surprised, and it makes it a little more difficult for me to review, as if that's the case that we intend the language to be different, it's tough to review for style, compared with our typical approach.

Additionally, I'm concerned that for an introduction, this document could be edited down in length, though I don't believe the PR conversation is the best place for that editing (line-by-line).

mDuo13 committed:
XRP Ledger Intro: revisions per reviews

MarkusTeufelberger commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -1,43 +1,52 @@
+This combines qualities of physical and centralized digital money. Prior to the invention of Bitcoin in 2009, all currencies could be divided into those two categories:
+
+- **Physical coins and paper money**, which individuals can use to do business without going through a central party. As physical objects, they cannot be used online, and doing business long-distance is slow and inconvenient.
+- **Centralized digital currencies**, which need an administrator to confirm transactions. The administrator also has the power to censor or roll back transactions, or disallow some individuals from using the digital currency. If company managing a digital currency decides someone has violated its terms of service, it can freeze or even confiscate that person's money. However, as digital balances ...
```
"If [the] company managing a digital currency" - maybe also change "company" to "operator", not all of these are run by companies.

MarkusTeufelberger commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -1,43 +1,52 @@
+
+**Note:** Users of the XRP Ledger _can_ freeze non-XRP currencies issued in the XRP Ledger. For more information, see the [Freeze documentation](concept-freeze.html).
 
 Furthermore, the XRP Ledger's system of trusted validators uses a small amount of human interaction to achieve better distribution of authority than other decentralized systems. Fully-automated systems for reaching consensus from an unknown set of participants are vulnerable to concentrations of voting power. For example, Bitcoin mining is disproportionately concentrated in places with cheap elec ...
```
Furthermore now sounds strange after the addition above.

MarkusTeufelberger commented on content/concept-xrp-ledger-intro.md:
```diff
@@ -65,19 +74,19 @@ Cryptography is one of the hardest parts of any distributed system, and a mistak
+[Modern Features for Smart Contracts]: #smart-features-for-smart-contracts
 
-Besides simple value transfer via XRP payments, the XRP Ledger has several advanced features that provide useful functions for building applications that use Internet of Value to serve previously unknown or impractical needs. Rather than running applications as "smart contracts" in the network itself, the XRP Ledger provides tools for contracts, while letting the applications themselves run anywh ...
+Besides simple value transfer with XRP payments, the XRP Ledger has several advanced features that provide useful functions for building applications that use the Internet of Value to serve previously unknown or impractical needs. Rather than running applications as "smart contracts" in the network itself, the XRP Ledger provides tools for contracts, while letting the applications themselves run  ...
```
Suggestion: "the XRP Ledger provides tools for [settling] contracts"?

mDuo13 commented:
(Rebased onto master to fix the broken `Amendments.cpp` link.)

I've addressed most of the feedback offered so far. And yes, @jbheron, with respect to the audience, this one is a bit different from some of the other docs in the developer portal. As an introductory document (sorry it's so long! there's a lot to introduce!) the goal is to explain where the XRP Ledger sits in the pantheon of technologies for people who're interested in it from a technical perspective, and to give them a basis for understanding the rest of the docs.

For me at least, the question I always want to know when learning about a new technology is, "Why?" So maybe the reason it sounds a bit more like a "marketing" document is an overlap there. I made an effort to highlight the things that are different, unusual, or unique about the XRP Ledger. Perhaps some things that are missing from an introductory document are a call to action, or a summary of the parts that _aren't_ unusual. I'd be interested to hear more ideas of things that the intro should, or shouldn't, spend time on.

mDuo13 committed:
XRP Ledger Intro: more fixes per review

mDuo13 committed:
Add XRP Ledger Intro ripple.com mapping
